### ISSUE TITLE
[1.16.4]-Do you have a plan?

### ISSUE THREAD

**Issue by EnterFor** [WesCook/Nutrition#207]
Are you planning to start 1.15 soon?
I love this Mod! Hope to play on 1.15
Sorry, I'm Chinese，English is very rotten

**Comment by WesCook** [comment]
I recently finished an update for 1.12 and will likely consider that branch completed.  It will enter maintenance mode now.

My plan is to update through the versions, so there's at least a Nutrition build available for each.  I probably won't offer very much (or maybe any) food support for each though.  I'd rather get the mod functional and leave that up to modpack developers.

Forge has marked 1.14 as an "LTS" version, but I don't really see why.  Moving to 1.15 makes more sense to me.  So 1.15 will likely become the new main development version (unless I get distracted and 1.16 comes out first).

**Comment by EnterFor** [comment]
> 我最近完成了1.12的更新，并可能会认为该分支已完成。现在它将进入维护模式。
> 
> 我的计划是更新各个版本，因此每个版本至少都有一个Nutrition版本。我可能不会为每个人提供太多（或可能有）食物支持。我宁愿让mod发挥功能并将其留给modpack开发人员使用。
> 
> Forge已将1.14标记为“ LTS”版本，但我真的不明白为什么。转到1.15对我来说更有意义。因此1.15可能会成为新的主要开发版本（除非我分心并且首先发布1.16）。

that's great. I'm looking forward to it! I'll keep an eye on it

**Comment by Mochnant** [comment]
Great news.  Love the mod.

**Comment by ghost** [comment]
> I recently finished an update for 1.12 and will likely consider that branch completed. It will enter maintenance mode now.
> 
> My plan is to update through the versions, so there's at least a Nutrition build available for each. I probably won't offer very much (or maybe any) food support for each though. I'd rather get the mod functional and leave that up to modpack developers.
> 
> Forge has marked 1.14 as an "LTS" version, but I don't really see why. Moving to 1.15 makes more sense to me. So 1.15 will likely become the new main development version (unless I get distracted and 1.16 comes out first).

are there any plans to port this to Fabric?

**Comment by WesCook** [comment]
No plans to port to Fabric.

**Comment by ghost** [comment]
Darn. Could I port it myself? (or at least a similar concept)

**Comment by WesCook** [comment]
Of course!  Nutrition was inspired by the nutrient-system in TerraFirmaCraft, after all.

**Comment by ghost** [comment]
Awesome! I'll totally try to work on that then.

**Comment by Mohatax995** [comment]
Any news on this? 1.15 is now marked as the LTS version instead of 1.14

**Comment by EnterFor** [comment]
Do you have plans for 1.16?

**Comment by WesCook** [comment]
Hi both,

After my previous comment in this issue, I did begin the port process.  I started a 1.14 branch back in May which is visible on GitHub.

I completed approximately 1/3rd of the port, but found there were significantly more changes required than I was expecting.  Virtually every subsystem had changed, including lifecycle events, capabilities, networking, commands, configs, GUIs, potions, metadata, and localization.  So essentially everything.

There weren't a lot of resources pertaining to an upgrade path, so there was a lot of "figure it out yourself" required at the time.  I ended up putting it down as a result, and haven't yet mustered up the will to work on the port again.

It's possible the process is easier now that 1.15 and 1.16 have started becoming more popular.  The 1.13/1.14 era was still pretty rough.  Against my earlier plans, I no longer believe porting to 1.14 makes much sense.  It was a poor pick for an LTS release, and there's no popular modpacks for that version.

I'm sorry to say that I have no immediate plans to resume progress.  That doesn't mean it won't happen, but there's nothing on the immediate horizon.  If I were to resume, I'd target 1.15 and 1.16.  But of course that can change too.

I hope you understand, and I thank you for your interest in the mod.

**Comment by Mohatax995** [comment]
Hi, perfectly understandable, I know its a hassle updating to 1.13+ so if you end up resuming the progress take your time. 
Thank you for making this great mod.

**Comment by choekstr** [comment]
These rapid fire releases really have hurt porting and any chance of an LTS version.  I thought 1.15, then 1.16, but 1.17 builds started coming out immediately and I can imagine most mod devs are feeling like "what's the point?". :(

I personally add nutrition to every single pack I play as it's such a great minigame and I can customize what effects and when for a tailored experience.  What a great mod and really hope you find the strength and desire to bring it forward (on forge of course :) ) so we can keep enjoying this on newer packs.  It looks like 1.16 even has a questing solution now so packs are starting to become competent again.

**Comment by lbowenwest** [comment]
Hi, I'm a huge fan of this mod and would love to see it in 1.16. Seeing as you have no current plans to continue porting, do you mind if I have a go? I'm happy to contribute the work back to this project, or if you don't like that to release it separately.

**Comment by WesCook** [comment]
Hi @lbowenwest.  You absolutely have my support if you'd like to attempt a port.  If it looks good I would accept a PR to merge it in for an official release.  Thanks for your interest, and I wish you the best of luck.

**Comment by Late-Night** [comment]
so, any progress on that 1.15 port?

**Comment by choekstr** [comment]
I saw it was forked about 12 days ago but no code has been checked in yet (probably learning about porting and testing and determining if 1.14 branch is worth leveraging or going straight from 1.12 to 1.16.  Fingers crossed lbowenwest can make some progress on the port.  Good Luck and we are here supporting you!

**Comment by lbowenwest** [comment]
So I haven't made a huge amount of progress, work picked up before the holidays, and I haven't done a whole lot recently. I've looked into how it all works under the hood, but my god there a lot of changes between 1.12 and 1.16 (guis were basically completely renamed for some reason and the effects and potions system seems to be pretty different as well which is helpful). I'm thinking it's probably worth just going straight to 1.16 and not worrying about 1.14

**Comment by choekstr** [comment]
@lbowenwest thanks on behalf of the community on providing an update, much appreciated!

That was my opinion as well, just skip straight to 1.16 as no one really plays the intermediary versions as no significant packs were released that use them.  it's like 1.9 and 1.11, no need to work hard to port mods for those MC versions.  Just my opinion of course.

**Comment by princeoofmash** [comment]
I have interesting in this mod since watched video to playing modpack in this. 
I hope to play on 1.7.10 but I know a backport to 1.7.10 was not planned.
Could I port  it myself?  
I'm japanese ,because I'm poor at English, I'm sorry if I am wrong.

**Comment by WesCook** [comment]
Hi @princeoofmash.  If you'd like to develop a fork of the mod for 1.7.10, you have my permission to do so.  The mod's license is MIT.

Let me know if you release it and I'll include a shoutout somewhere.

**Comment by princeoofmash** [comment]
Thank you!!
I'll definitely contact you again ,when this mod get completed.

**Comment by ghost** [comment]
I don't know how to do proper pull requests and stuff like that but if i could probably would help port it to 1.17, just waiting for that release + the forge for it.

**Comment by WesCook** [comment]
As development has slowed and newer versions of Minecraft become more widely adopted, I think it's best to consider Nutrition in a maintenance state now.  I'll continue to fix bugs and do a food update passes now and then for 1.12 packs, but new features and big changes are unlikely.

While I wouldn't say there's zero probability of a future port, I think I missed the window when it made the most sense.  As such I'll go ahead and close this issue now.

I've had a few requests from developers wishing to port the mod, and I give full permission if anybody wishes to undertake that task.  My only request is that you not use the same name (eg. "Nutrition 2") as to prevent confusion.  If the mod is successfully ported, I'd also be open to a dual maintainer relationship if you'd like it to be an official continuation of Nutrition.

For 1.16 a number of mods have popped up which fill the same niche.  I'd like to specifically call out [Nutritional Balance](https://www.curseforge.com/minecraft/mc-mods/nutritional-balance), whose author graciously reached out to me before release.  Other alternatives are [Diet](https://www.curseforge.com/minecraft/mc-mods/diet) for Forge, and [Nourish](https://www.curseforge.com/minecraft/mc-mods/nourish) for Fabric.  Feel free to share other alternatives below.

I'd like to thank everybody for taking an interest in the mod.  Nutrition has been a project dear to my heart, and I really appreciate all the community support it's received.  Thanks everyone!

**Comment by mitchej123** [comment]
Piggybacking on this thread - We've made an unofficial backport to 1.7.10, with an initial [0.0.1 release](https://github.com/GTNewHorizons/Nutrition/releases/tag/0.0.1) available. We'll be updating the documentation and the features in the weeks to come.  Thanks to @WesCook for the original mod which I loved on newer MC versions!

**Comment by WesCook** [comment]
Hey Jason,

That's awesome, well done to you and Dane.  I can confirm that I gave you permission in January of 2022 to port the mod to 1.7.10.  Of course, the license would permit it regardless, but I appreciate you asking all the same.

I had a quick read through the new commits.  One thing that struck me is that in many ways, older Minecraft was simpler.  Fewer abstractions, less complication.

The funny thing is, the PlayerDataHandler class (your replacement for Capabilities) is how I wrote the mod originally.  I was storing everything in a server-side class, and just needed a way to save that data to the game's world file.  I couldn't find any way to do that, so I joined the Forge Discord and learned I needed to use this new thing called "Capabilities".  Ultimately had to rewrite the whole system.

Anyway, I'm glad you were able to make use of the project for GT:NH.  That's a modpack I've always been curious about, but never brave enough to try.  I'm definitely interested in taking a peak now though to see how it all works together.

Thanks for your work on this, and for sharing the update here.  Looking forward to seeing what else you do with it.

And by the way, if Dane happened to spot any bugs that are applicable to the 1.12 version, let me know!  I'd likely still fix them, even today.

Cheers.
